androidApp/src/main/java/com/example/tasky/android/agenda/components/AgendaDayBar.kt

```diff
@@ -0,0 +1,57 @@
+package com.example.tasky.android.agenda.components
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import com.example.tasky.android.theme.MyApplicationTheme
+import kotlinx.collections.immutable.ImmutableList
+import kotlinx.collections.immutable.toImmutableList
+import kotlinx.datetime.Clock
+import kotlinx.datetime.DateTimeUnit
+import kotlinx.datetime.Instant
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.plus
+import kotlinx.datetime.toLocalDateTime
+
+@Composable
+fun AgendaDayBar(
+    days: ImmutableList<Instant>,
+    selectedDayOffset: Int,
+    onDaySelect: (Int) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Row(modifier = modifier, horizontalArrangement = Arrangement.SpaceBetween) {
+        days.mapIndexed { index, day ->
+            val localDate = day.toLocalDateTime(TimeZone.currentSystemDefault())
+            AgendaDayItem(
+                upperString =
+                    localDate.dayOfWeek.name
+                        .first()
+                        .toString(),
+                lowerString = localDate.dayOfMonth.toString(),
+                index == selectedDayOffset,
+                onClick = { onDaySelect(index) },
+            )
+        }
+    }
```

Comment:
This looks good on your device, but that doesn't mean it looks good on any device. This code works if all days can be fit in the screen width, but if not, they will overlap and not be scrollable. Could it therefore make sense to work with a LazyRow instead, to also support showing more days on larger screens?

Reply:
but the behaviour is not the same, using LazyRow, I need to add a fixed padding between them instead of evenly distrubuted space according to screen width, which would cause the row not spanning the whole width in some screen sizes.

Reply:
You can set the `LazyRow` to fill the screen to the maximum width. It is also fine to provide padding values for each item.

Reply:
You just need to make sure that you're having enough days in your row. I personally calculate the past 15 and next 15 days to have 30 days in total. That should fit on any device I'd say 🤔